src/web/mjs/connectors/Pururin.mjs

```diff
@@ -1,31 +1,83 @@
-
-    /**
-     *
-     */
     constructor() {
         super();
         super.id = 'pururin';
         super.label = 'Pururin';
-        this.tags = [];
-        this.url = 'https://pururin.io';
-        this.links = {
-            login: 'https://pururin.io/login'
-        };
+        this.tags = ['manga', 'hentai', 'english'];
+        this.url = 'https://pururin.to';
+        this.CDN = "https://cdn.pururin.to/assets/images/data/{1}/{2}.{3}";
+        this.path = "/browse/title";
+        this.api = "/api/contribute/gallery/info";
+        this.queryMangasPageCount = 'ul.pagination.flex-wrap li:nth-last-of-type(2) a';
+        this.queryMangas = 'a.card.card-gallery';
+        this.queryChapters ='div.gallery-action a';
+        this.requestOptions.headers.set('x-referer', this.url);
+        this.requestOptions.headers.set('x-origin', this.url);
+    }
+    async _getMangas() {
+        let mangaList = [];
+        const uri = new URL(this.path, this.url);
+        const request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryMangasPageCount);
+        let pageCount = parseInt(data[0].text);
+        for(let page = 1; page <= pageCount; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+    async _getMangasFromPage(page) {
+        let uri = new URL(this.path + '?page='+page, this.url);
+        let request = new Request(uri, this.requestOptions);
+        let data = await this.fetchDOM(request, this.queryMangas);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                title: element.querySelector('source.card-img-top').getAttribute('alt').trim()
+            };
+        });
     }
-
-    _getMangaList( callback ) {
-        callback( new Error( 'Please report this broken website on HakuNeko\'s GitHub project page.' ), undefined );
+    async _getChapters(manga) {
+        let mid = this.getRootRelativeOrAbsoluteLink(manga.id, this.url);
+        const chapters = [{ id: mid, title: 'Chapter' }];
+        return chapters;
     }
-    _getChapterList( manga, callback ) {
-        callback( new Error( 'Please report this broken website on HakuNeko\'s GitHub project page.' ), undefined );
+    async _getPages(chapter) {
+        let mangaID = chapter.id.match(/\/gallery\/([0-9]+)/)[1];
+        const uri = new URL(this.api, this.url);
+        const request = this.getApiRequest(uri, mangaID);
+        const data = await this.fetchJSON(request);
+        let pagesMax = data.gallery.total_pages;
+        let extension = data.gallery.image_extension;
+        //https://cdn.pururin.to/assets/images/data/<mangaid>/<i>.image_extension
+        return new Array(pagesMax).fill().map((_, index) => new URL(this.CDN.replace('{1}', mangaID).replace('{2}', index+1).replace('{3}', extension)).href);
```

Comment:
May change CDN, so string interpolation can be used instead of `replace` chaining
```javascript
`${this.CDN}/${mangaID}/${index + 1}.${extension}`
```